packages/calypso-e2e/src/element-helper.ts

```diff
@@ -41,41 +47,54 @@ export async function waitForElementEnabled(
 /**
  * Locates and clicks on a specified tab on the NavTab.
  *
- * NavTabs are used throughout calypso to contain multiple related but separate pages within the same
- * overall page. For instance, on the Media gallery page a NavTab is used to filter the gallery to
+ * NavTabs are used throughout calypso to contain sub-pages within the parent page.
+ * For instance, on the Media gallery page a NavTab is used to filter the gallery to
  * show a specific category of gallery items.
  *
  * @param {Page} page Underlying page on which interactions take place.
  * @param {string} name Name of the tab to be clicked.
+ * @throws {Error} If the tab name is not the active tab.
  */
 export async function clickNavTab( page: Page, name: string ): Promise< void > {
-	// Mobile view - navtabs become a dropdown.
+	const selectedTabLocator = page.locator( selectors.navTabItem( { selected: true } ) );
+	const selectedTabName = await selectedTabLocator.innerText();
+
+	// Short circuit operation if the active tab and target tabs are the same.
+	if ( selectedTabName === name ) {
+		return;
+	}
+
+	// Mobile view - navtabs become a dropdown and thus it must be opened first.
 	if ( envVariables.VIEWPORT_NAME === 'mobile' ) {
-		await page.waitForLoadState( 'networkidle' );
-		const navTabsButtonLocator = page.locator( selectors.mobileNavTabsToggle );
+		// Layout can shift in certain pages (eg. PlansPage) as elements are lazy loaded.
+		const mainElement = await page.waitForSelector( selectors.main );
+		await Promise.all( [
+			page.waitForLoadState( 'networkidle' ),
+			mainElement.waitForElementState( 'stable' ),
+		] );
```

Comment:
So, I've spent a bit of time debugging this, and I'm not entirely sure that waiting for the 'stable' animation is going to _always_ stop this failure, although I think it will help.

I think Louis's comment here: p1654583626042029/1654264833.113049-slack-C1A1EKDGQ is explaining what is going on. There's a react lifecycle method that's firing at the end of everything that's closing the mobile pane.

I've watched this page load throttled several times, and here's what you usually see:
1. The top nav toolbar loads in with a loading indicator underneath it.
2. All the page plan information loads, but the SVGs/icons haven't loaded yet.
3. The last thing to happen is all the SVG/icons load in. ---> this is when you see the react lifecycle method fire too, and the mobile pane close.

The catch is that, unless I'm misunderstanding the docs, [all that "stable" is looking for under the hood](https://playwright.dev/docs/actionability#stable) is just for the bounding rectangle of an element to be the same dimensions over two consecutive re-paints/re-render frames.

The catch is that, from what I can tell playing around in the Chrome console, that's true for several frames _before_ that final load happens. Especially with how this is called in `Promise.all`, this happens several times right as the page loads, so that part of the promise will likely resolve really early.

Moving that stable check out of the `Promise.all` might slow things down enough that we'll be less likely to see the failure, but theoretically, it probably could still happen.

Since this is unique to the plans page, I wonder if we might be best off just putting in a one-off kind of hack for that page until the lifecycle issue is fixed?

Since the very last thing that seems to happen is that the Plan svg's load in, maybe we could do a wait for one of those responses in the POM for the plans page?

```typescript
await page.waitForResponse(response => response.url().includes("/images/wpcom-ecommerce"));
```

It's hacky, but I think it might be even more reliable, and it lets us isolate the hacky-ness to the single page where this is a problem.

I'm definitely not married to that idea though!

Thoughts?

Reply:
Yes, certainly good points!

> [all that "stable" is looking for under the hood](https://playwright.dev/docs/actionability#stable) is just for the bounding rectangle of an element to be the same dimensions over two consecutive re-paints/re-render frames...this happens several times right as the page loads, so that part of the promise will likely resolve really early

You are 100% correct. I have often resorted to using `stable` (especially in the early days of the framework) to try to deal with animations, but it's often an incomplete workaround for the reasons you state here.

If you recall, the lazy-loading upsell banner in the `SidebarComponent` also suffers from the same problem. From Playwright's perspective, the `SidebarComponent` is already stable even though in reality the upsell banner is just about to destroy this man's whole automation flow.

> Moving that stable check out of the Promise.all might slow things down enough that we'll be less likely to see the failure, but theoretically, it probably could still happen.

Yeah, this would just end up being akin to a variable version of the `page.waitForTimeout` which is sketchy at best.

> Since this is unique to the plans page, I wonder if we might be best off just putting in a one-off kind of hack for that page until the lifecycle issue is fixed?

As much as you and (and everyone maintaining this down the road) will hate it, this may be the only option we have.

I think rest of the changes to `ElementHelper.clickTab` is worth keeping, so my proposal is this:
- remove the additional wait on `main` to be stable (restore the initial condition, which was only a `networkidle`);
- in `PlansPage.clickTab`, add a specific wait for an SVG as you suggest.

Reply:
Pushed up f506b0d478179945ff25371efe1bf27f157b2d13 which implements the suggestions.